src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -49,7 +50,11 @@ public Task UnloadAsync()
 
         protected override Task DisposeCoreAsync(bool initialized)
         {
-            s_listOfMissingRuntimes.Clear();
+            lock (s_listOfMissingRuntimesLock)
```

Comment:
This component lives per project, yet clears a solution-wide collection?

Reply:
It checks all projects, but if all projects are missing Runtime 5.0 it will only register once in the solution.

Reply:
As in, let's pretend this solution has 500 projects. This lock will get hit 500 times to attempt to clear a collection that is already empty after the first call. Why do we need the other 499 calls?

Reply:
Yes this is broken. This was likely trying to clean up on solution close, but it'll do the wrong thing when a project is unloaded or reloaded, for example.

It looks like this component is trying to de-duplicate data for `IMissingSetupComponentRegistrationService`. That component should do its own de-duplication, and all state in this `MissingSdkRuntimeDetector` should be tied to its own lifetime.

Reply:
@drewnoakes I did the suggested changes. 
The de-duplication was moved to `MissingSdkRuntimeDetector`